### PR TITLE
Analyzer remove unused projection columns

### PR DESCRIPTION
**Pull request by kitaisreal**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Analyzer/QueryNode.h [hunk 1]
```diff
@@ -556,10 +556,13 @@ class QueryNode final : public IQueryTreeNode
     }
 
     /// Resolve query node projection columns
-    void resolveProjectionColumns(NamesAndTypes projection_columns_value)
-    {
-        projection_columns = std::move(projection_columns_value);
-    }
+    void resolveProjectionColumns(NamesAndTypes projection_columns_value);
```
Comment (kitaisreal) [inline src/Analyzer/QueryNode.h:559]: I will add refactoring of `QueryNode` and `UnionNode` to have `QueryNodeBase` parent class and extract these methods and other common methods into this parent class.

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 3d54dea786f3e51d23e506bd4bdfaae193850f48 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56499/3d54dea786f3e51d23e506bd4bdfaae193850f48/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6810037926">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56499/3d54dea786f3e51d23e506bd4bdfaae193850f48/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by KochetovNicolai — APPROVED** [review]

**Comment by KochetovNicolai** [comment]
Some tests with analyzer are failed. We need to fix it.

File: src/Analyzer/QueryNode.h (hunk 1 above)
Comment (novikd) [reply src/Analyzer/QueryNode.h:559]: We need to do the same with `TableNode` and `TableFunctionNode`

**Review by novikd — APPROVED** [review]
LGTM

File: tests/analyzer_tech_debt.txt [hunk 2]
```diff
@@ -11,12 +10,9 @@
 01244_optimize_distributed_group_by_sharding_key
 01268_mv_scalars
 01268_shard_avgweighted
-01287_max_execution_speed
```
Comment (Algunenano) [inline tests/analyzer_tech_debt.txt:14]: No matter how many times this test is removed from the list, it's flaky with the analyzer and the analyzer only and it needs fixing. See https://s3.amazonaws.com/clickhouse-test-reports/56984/ff438112e95568bb322c9ff8283e6b7f2e1b412a/stateless_tests__release__analyzer_.html

5 failures in 24 hours. All with the analyzer

**Pull request merged**